subsys/sal/sid_pal/src/sid_pal_serial_bus_spi.c

```diff
@@ -30,13 +30,21 @@ static sid_error_t bus_serial_spi_xfer(const struct sid_pal_serial_bus_iface *if
 				       size_t xfer_size);
 static sid_error_t bus_serial_spi_destroy(const struct sid_pal_serial_bus_iface *iface);
 
-static struct bus_serial_ctx_t bus_serial_ctx;
-
 static const struct sid_pal_serial_bus_iface bus_ops = {
 	.xfer = bus_serial_spi_xfer,
 	.destroy = bus_serial_spi_destroy,
 };
 
+static struct bus_serial_ctx_t bus_serial_ctx = {
+	.iface = &bus_ops,
+	.device = DEVICE_DT_GET(DT_NODELABEL(sid_semtech)),
+	.cfg = (struct spi_config){
+		.frequency = DT_PROP_OR(DT_NODELABEL(sid_semtech), clock_frequency, SPI_FREQUENCY_DEFAULT),
```

Comment:
Use `bus_selector.speed_hz` because in subsys/config/common/src/app_subGHz_config.c file you get frequency value from dts.

Reply:
ok, but it will have to be in `sid_pal_serial_bus_nordic_spi_create`, because it is not static value.

But generally I agree, that if we already got this value in one place, any other place it occurs should be a reference to the original place that it has been defined